opensrp-opd/src/main/java/org/smartregister/opd/repository/OpdDiagnosisDetailRepository.java

```diff
@@ -0,0 +1,75 @@
+            + OpdDbConstants.Column.OpdDiagnosisDetail.DIAGNOSIS_SAME + " VARCHAR NULL, "
+            + OpdDbConstants.Column.OpdDiagnosisDetail.ICD10_CODE + " VARCHAR NULL, "
+            + OpdDbConstants.Column.OpdDiagnosisDetail.CODE + " VARCHAR NULL, "
+            + OpdDbConstants.Column.OpdDiagnosisDetail.DETAILS + " VARCHAR NULL, "
+            + OpdDbConstants.Column.OpdDiagnosisDetail.CREATED_AT + " VARCHAR NULL, "
+            + OpdDbConstants.Column.OpdDiagnosisDetail.VISIT_ID + " VARCHAR NOT NULL )";
+
+    private static final String INDEX_BASE_ENTITY_ID = "CREATE INDEX " + OpdDbConstants.Table.OPD_DIAGNOSIS_DETAIL
+            + "_" + OpdDbConstants.Column.OpdDiagnosisDetail.BASE_ENTITY_ID + "_index ON " + OpdDbConstants.Table.OPD_DIAGNOSIS_DETAIL +
+            "(" + OpdDbConstants.Column.OpdDiagnosisDetail.BASE_ENTITY_ID + " COLLATE NOCASE);";
+
+    private static final String INDEX_VISIT_ID = "CREATE INDEX " + OpdDbConstants.Table.OPD_DIAGNOSIS_DETAIL
+            + "_" + OpdDbConstants.Column.OpdDiagnosisDetail.VISIT_ID + "_index ON " + OpdDbConstants.Table.OPD_DIAGNOSIS_DETAIL +
+            "(" + OpdDbConstants.Column.OpdDiagnosisDetail.VISIT_ID + " COLLATE NOCASE);";
+
+    public static void createTable(@NonNull SQLiteDatabase database) {
+        database.execSQL(CREATE_TABLE_SQL);
+        database.execSQL(INDEX_BASE_ENTITY_ID);
+        database.execSQL(INDEX_VISIT_ID);
+    }
+
+    @Override
+    public boolean saveOrUpdate(@NonNull org.smartregister.opd.pojo.OpdDiagnosisDetail opdDiagnosisDetail) {
+        ContentValues contentValues = new ContentValues();
+        contentValues.put(OpdDbConstants.Column.OpdDiagnosisDetail.BASE_ENTITY_ID, opdDiagnosisDetail.getBaseEntityId());
+        contentValues.put(OpdDbConstants.Column.OpdDiagnosisDetail.DIAGNOSIS, opdDiagnosisDetail.getDiagnosis());
+        contentValues.put(OpdDbConstants.Column.OpdDiagnosisDetail.TYPE, opdDiagnosisDetail.getType());
+        contentValues.put(OpdDbConstants.Column.OpdDiagnosisDetail.DISEASE, opdDiagnosisDetail.getDisease());
+        contentValues.put(OpdDbConstants.Column.OpdDiagnosisDetail.ICD10_CODE, opdDiagnosisDetail.getIcd10Code());
+        contentValues.put(OpdDbConstants.Column.OpdDiagnosisDetail.CODE, opdDiagnosisDetail.getCode());
+        contentValues.put(OpdDbConstants.Column.OpdDiagnosisDetail.DIAGNOSIS_SAME, opdDiagnosisDetail.getIsDiagnosisSame());
+        contentValues.put(OpdDbConstants.Column.OpdDiagnosisDetail.DETAILS, opdDiagnosisDetail.getDetails());
+        contentValues.put(OpdDbConstants.Column.OpdDiagnosisDetail.VISIT_ID, opdDiagnosisDetail.getVisitId());
+        contentValues.put(OpdDbConstants.Column.OpdDiagnosisDetail.CREATED_AT, opdDiagnosisDetail.getCreatedAt());
+        SQLiteDatabase sqLiteDatabase = getWritableDatabase();
+        long rows = sqLiteDatabase.insert(OpdDbConstants.Table.OPD_DIAGNOSIS_DETAIL, null, contentValues);
```

Comment:
I am not sure if `insert` will add or update, does it?

Reply:
No it doesn't but am updating the function to only say save, so that there can be no confusion.